src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PrivateDataRetrievalService.java

```diff
@@ -1,125 +1,242 @@
+            ApiLogger.info("Retrieving overseas entity data for company number ", logMap);
+
+            return overseasEntityDataApi;
+        } catch (ApiErrorResponseException e) {
+            if (e.getStatusCode() == HttpServletResponse.SC_NOT_FOUND) {
+                ApiLogger.info("No overseas entity data for Company Number " + companyNumber,
+                        logMap);
+                return new OverseasEntityDataApi();
+            }
+            throw new ServiceException(e.getStatusMessage(), e);
+        } catch (URIValidationException e) {
+            var message = "Error Retrieving overseas entity data for " + companyNumber;
+            ApiLogger.error(message, e, logMap);
+            throw new ServiceException(e.getMessage(), e);
+        }
+    }
+
+    public PrivateBoDataListApi getBeneficialOwnersData(String companyNumber)
+            throws ServiceException {
+
+        var logMap = new HashMap<String, Object>();
+        logMap.put(COMPANY_NUMBER, companyNumber);
+        ApiLogger.info("Retrieving Beneficial Owners for Company Number " + companyNumber, logMap);
+
+        try {
+            PrivateBoDataListApi beneficialOwnerDataList = apiClientService.getInternalApiClient()
+                    .privateBeneficialOwnerResourceHandler()
+                    .getBeneficialOwners( OVERSEAS_ENTITY_URI_SECTION + companyNumber + "/beneficial-owners")
+                    .execute()
+                    .getData();
+
+            if (beneficialOwnerDataList != null
+                    && beneficialOwnerDataList.getBoPrivateData() != null
+                    && !beneficialOwnerDataList.getBoPrivateData().isEmpty()) {
+                int numberOfBOs = beneficialOwnerDataList.getBoPrivateData().size();
+                ApiLogger.info(String.format("Retrieved %d Beneficial Owners for Company Number %s",
+                        numberOfBOs, companyNumber));
+            }
+
+            return beneficialOwnerDataList;
+        } catch (ApiErrorResponseException e) {
+            if (e.getStatusCode() == HttpServletResponse.SC_NOT_FOUND) {
+                ApiLogger.info("No Beneficial Owners found for Company Number " + companyNumber,
+                        logMap);
+                return new PrivateBoDataListApi(Collections.emptyList());
+            }
+            throw new ServiceException(e.getStatusMessage(), e);
+        } catch (URIValidationException e) {
+            var message = "Error Retrieving Beneficial Owner data for " + companyNumber;
+            ApiLogger.errorContext(message, e);
+            throw new ServiceException(e.getMessage(), e);
+        }
+    }
+
+    public PrivateCorporateTrusteeListApi getCorporateTrustees(String hashedTrustId,
+            String companyNumber) throws ServiceException {
+
+        var logMap = new HashMap<String, Object>();
+
+        String nonHashedId = findMatchingId(hashedTrustId, companyNumber);
+
+        logMap.put(TRUST_ID, nonHashedId);
```

Comment:
move to line 148 just to keep log stuff together

Reply:
That won't work as it needs `nonHashedId`

Reply:
True - could add logMap.put(COMPANY_NUMER, companyNumber) as well?